stories/browser/src/TransactionDetails.tsx

```diff
@@ -0,0 +1,204 @@
+        return null;
+    }
+    return (
+        <View style={styles.parameter}>
+            <UILabel role={TypographyVariants.ParagraphLabel} color={UILabelColors.TextSecondary}>
+                {label}
+            </UILabel>
+            <UIPressableArea onPress={onPress} disabled={!onPress} style={styles.parameterValue}>
+                <UILabel role={TypographyVariants.ParagraphText}>
+                    {prefix}
+                    <UICurrency signChar={signChar} decimalAspect={UINumberDecimalAspect.Precision}>
+                        {value}
+                    </UICurrency>
+                </UILabel>
+            </UIPressableArea>
+        </View>
+    );
+}
+
+function ExpandedContent(props: TransactionDetailsProps) {
+    const {
+        signature,
+        action,
+        recipient,
+        amount,
+        amountCurrency,
+        contractFee,
+        contractFeeCurrency,
+        networkFee,
+        networkFeeCurrency,
+        onRecipientPress,
+    } = props;
+    return (
+        <>
+            <TextParameter
+                label={uiLocalized.Browser.TransactionConfirmation.Signature}
+                value={signature?.title}
+            />
+            <TextParameter
+                label={uiLocalized.Browser.TransactionConfirmation.Action}
+                value={action}
+            />
+            <TextParameter
+                label={uiLocalized.Browser.TransactionConfirmation.Recipient}
+                value={recipient}
+                onPress={onRecipientPress}
+            />
+            <NumberParameter
+                label={uiLocalized.Browser.TransactionConfirmation.Amount}
+                value={amount}
+                signChar={amountCurrency}
+            />
+            <NumberParameter
+                label={uiLocalized.Browser.TransactionConfirmation.ContractFee}
+                value={contractFee}
+                signChar={contractFeeCurrency}
+                prefix={`${uiLocalized.Browser.TransactionConfirmation.UpTo} `}
+            />
+            <NumberParameter
+                label={uiLocalized.Browser.TransactionConfirmation.NetworkFee}
+                value={networkFee}
+                signChar={networkFeeCurrency}
+                prefix="≈"
+            />
+        </>
+    );
+}
+
+function FoldedContent(props: TransactionDetailsProps) {
+    const { amount, amountCurrency, contractFee, contractFeeCurrency } = props;
+    return (
+        <>
+            <NumberParameter
+                label={uiLocalized.Browser.TransactionConfirmation.Amount}
+                value={amount}
+                signChar={amountCurrency}
+            />
+            <NumberParameter
+                label={uiLocalized.Browser.TransactionConfirmation.ContractFee}
+                value={contractFee}
+                signChar={contractFeeCurrency}
+                prefix={`${uiLocalized.Browser.TransactionConfirmation.UpTo} `}
+            />
+        </>
+    );
+}
```

Comment:
Are we really need split components?

Reply:
Yes. It brings us a simple and readable code. And protect from layout animations bugs.